benchmark/solver/solver.cpp

```diff
@@ -406,7 +409,8 @@ void solve_system(const std::string& solver_name,
         for (auto _ : ic.warmup_run()) {
             auto x_clone = clone(x);
             auto precond = precond_factory.at(precond_name)(exec);
-            auto solver = generate_solver(exec, give(precond), solver_name)
+            auto solver = generate_solver(exec, give(precond), solver_name,
+                                          FLAGS_warmup_max_iters)
```

Comment:
We are writing the number of iterations out from it_logger, this may give us too small iteration counts. Maybe we should combine this with the ResidualLogger run below?

Reply:
I actually encountered a related issue in #750 where the iteration count is not available if we don't do warmup iterations.